internal/adapters/primary/http/handlers.go

```diff
@@ -0,0 +1,170 @@
+}
+
+func NewHandler(authService ports.AuthService) *Handler {
+	return &Handler{
+		authService: authService,
+	}
+}
+
+type registerRequest struct {
+	Email     string `json:"email"`
+	Password  string `json:"password"`
+	FirstName string `json:"first_name"`
+	LastName  string `json:"last_name"`
+}
+
+type loginRequest struct {
+	Email    string `json:"email"`
+	Password string `json:"password"`
+}
+
+// type resetPasswordRequest struct {
+// 	Token       string `json:"token"`
+// 	NewPassword string `json:"new_password"`
+// }
+
+// type changePasswordRequest struct {
+// 	OldPassword string `json:"old_password"`
+// 	NewPassword string `json:"new_password"`
+// }
+
+func (h *Handler) RegisterHandler(w http.ResponseWriter, r *http.Request) {
+	var req registerRequest
+	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
+		http.Error(w, "Invalid request body", http.StatusBadRequest)
+		return
+	}
+
+	// Validate required fields
+	if req.Email == "" || req.Password == "" || req.FirstName == "" || req.LastName == "" {
+		http.Error(w, "All fields are required", http.StatusBadRequest)
+		return
+	}
+
+	domainReq := domain.RegisterRequest{
+		Email:     req.Email,
+		Password:  req.Password,
+		FirstName: req.FirstName,
+		LastName:  req.LastName,
+	}
+
+	user, err := h.authService.Register(domainReq)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusCreated)
+	if err := json.NewEncoder(w).Encode(user); err != nil {
+		http.Error(w, "Failed to encode response", http.StatusInternalServerError)
+		return
+	}
+}
+
+func (h *Handler) LoginHandler(w http.ResponseWriter, r *http.Request) {
+	var req loginRequest
+	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
+		http.Error(w, "Invalid request body", http.StatusBadRequest)
+		return
+	}
+
+	// Validate required fields
+	if req.Email == "" || req.Password == "" {
+		http.Error(w, "Email and password are required", http.StatusBadRequest)
+		return
+	}
+
+	domainReq := domain.LoginRequest{
+		Email:    req.Email,
+		Password: req.Password,
+	}
+
+	token, err := h.authService.Login(domainReq)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusUnauthorized)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusOK)
+	if err := json.NewEncoder(w).Encode(map[string]string{"token": token}); err != nil {
+		http.Error(w, "Failed to encode response", http.StatusInternalServerError)
+		return
+	}
+}
+
+// func (h *Handler) VerifyEmailHandler(w http.ResponseWriter, r *http.Request) {
+// 	token := r.URL.Query().Get("token")
+// 	if token == "" {
+// 		http.Error(w, "Missing token", http.StatusBadRequest)
+// 		return
+// 	}
+
+// 	if err := h.authService.VerifyEmail(token); err != nil {
+// 		http.Error(w, err.Error(), http.StatusBadRequest)
+// 		return
+// 	}
+
+// 	w.WriteHeader(http.StatusOK)
+// }
```

Comment:
Don't include commented out code in an MR ready for review, find some other way of holding on to code you might need later